server/src/infra/configuration.rs

```diff
@@ -136,6 +156,8 @@ pub struct Configuration {
     #[serde(skip)]
     #[builder(field(private), default = "None")]
     server_setup: Option<ServerSetupConfig>,
+    #[builder(default = r#"Vec::new()"#)]
+    pub plugin: Vec<PluginOptions>,
```

Comment:
How does that work with env variables? Currently, every LLDAP setting is also available as an env variable.

I propose a different structure:
```toml
[[plugins.pam_plugin]]
enabled = true
plugin_path = "plugins/pam.lua"
[[plugins.pam_plugin.config]]
uid_offset = "10000"
```

With some checks that the loaded plugin declares the same name as the config.

Reply:
That's perfectly fine with me, honestly expected you to have a strong opinion here, so haven't given this too much attention. :-)

What about just "plugins.pam" though, instead of "plugins.pam_plugin"? Latter seems a bit redundant, and the former fits a bit more with what the plugin would declare as well?

Reply:
Yeah, `pam_plugin` was my placeholder for the plugin name. It would be `plugins.pam` in your case.

Reply:
Restructured as discussed.

Are we sure, we'd like to enforce that the same name is used as the plugin is declaring? It would deny the possibility of declaring a plugin twice, which might be interesting for a web-hook plugin (with different configs, I'd assume). It's still possible to reliably provide setup instructions using env vars without us forcing it.